### ISSUE TITLE
Output when execution fails

### ISSUE THREAD

**Issue by filhodanuvem** [grafana/xk6-exec#12]
I think it would be helpful to have a way to see/get the output of failure executions, from what I see we don't have it because the `log.fatal` [here](https://github.com/grafana/xk6-exec/blob/main/exec.go#L56) exits the run. 
What do you think?

**Comment by javaducky** [comment]
Hi @filhodanuvem! Thank you for noticing that...it's not much help when the command silently fails.

**Comment by diegombeltran** [comment]
+1 to this feature!

**Comment by fczuardi** [comment]
What would be the best return for the exec.command in this case? Log the error and return an empty string? Log the error and return the exit status as a string? Or return the same string that is being printed in the log.fatal (but without [interrupting the main execution](https://github.com/grafana/xk6-exec/issues/17)?

**Comment by fczuardi** [comment]
Out of curiosity, I went looking for the [forks of this project](https://github.com/grafana/xk6-exec/forks?include=active&page=1&period=&sort_by=stargazer_counts) and it seems that changing this is a popular motivator forforking.

Some replace this `log.Fatal` with a `log.Print`, like https://github.com/grafana/xk6-exec/compare/main...eznd-go:xk6-exec:main and and https://github.com/grafana/xk6-exec/compare/main...fcurella:xk6-exec:main while other even put it under an option for backwards compatibility: https://github.com/grafana/xk6-exec/commit/60c5a41c8c256d2505745a007d5941b964f6e6c7

It would be nice to have one of the patches landed on the main project. Is there any reason not to do it? What is missing? The pull requests?

**Comment by filhodanuvem** [comment]
In my opinion is just about create the pull request, time has passed and I didn't do it. Go for it @fczuardi if that unblocks you in any way. 
The best approach for me would be changing the method `Command` to return also an error, the caller would be responsible to panic it or not but I understand that this might generates a big PR. 

https://github.com/grafana/xk6-exec/blob/997ee1bd7ecd0be67bdb8f803c9683f465194f76/exec.go#L49-L58

**Comment by romaia** [comment]
So, this was bothering me as well, so I created PR #20 that allows to specify an option `fatalError: false` to disable the current behavior. When ommited, it will keep failing for backwards compatibility.

I also took the oportunity to add support for an `env` option that will set environment variables.

**Comment by mark-00** [comment]
I did some research on the topic. As I understand it, goja only exposes the err in the javascript exception. 
err is of type https://pkg.go.dev/os/exec#ExitError so we have to extend the returned error object to include stdout
goja should automaticaly pick that up
Can somebody check if this is feasable/correct ?


```
func (*EXEC) Command(name string, args []string, option CommandOptions) (string, error) {
	cmd := exec.Command(name, args...)
	if option.Dir != "" {
		cmd.Dir = option.Dir
	}

	out, err := cmd.Output()
	if err != nil && !option.ContinueOnError {
		log.Fatal(err.Error() + " on command: " + name + " " + strings.Join(args, " "))
	}

	if err !=nil  && option.IncludeSdtoutOnError
	    return string(out), { os.ProcessState: err.os.ProcessState, Stderr : err.Stderr, Stdout out}
    }   

	return string(out), err
}

```

In the k6 script this then can be used in a fully backward compatible way
```
        try {
		result = exec.command('ls', ["filename"], { "continue_on_error": true }, {"include_stdout_on_error": true});
	 } catch (e) {
		console.error(e);
		console.error(e.toString());
		console.error(String.fromCharCode(...e.value.stderr));  //this already works
		console.error(String.fromCharCode(...e.value.stdout));  //this is new
		//this works to see https://pkg.go.dev/os#ProcessState
		console.error(e.value.process_state.exitCode());
		console.error(e.value.process_state.userTime());
		console.error(e.value.process_state.pid());
		console.error(e.value.process_state.string()  );
		console.error(e.value.process_state.toString());
	 }
 ```

**Comment by mark-00** [comment]
Turns out it is feasable https://github.com/grafana/xk6-exec/pull/29

**Comment by mstoykov** [comment]
We are archiving this repository due to the team not intending on maintaing it more - if you want this functionality - you will need to make a fork or use one already made
